doc/usr/source/9_other/11_minimal_correction_set.rst

```diff
@@ -0,0 +1,65 @@
+.. _9_other/11_minimal_correction_set:
+
+Minimal Cut Set
+================
+
+The minimal cut set generation is a special mode where Kind 2 computes a minimal subset of
+the model elements (contract items, equations or node calls) such that the system become unsafe if we remove these assumptions.
+
+To enable minimal cut set generation, run
+
+.. code-block:: none
+
+  kind2 <lustre_file> --enable MCS
+
+Options
+-------
+
+* ``--mcs_category {annotations|node_calls|contracts|equations|assertions}`` (default: annotations) -- Consider only a specific category of elements, repeat option to consider multiple categories
+* ``--mcs_only_main_node <bool>`` (default ``false``\ ) -- Compute a MCS over the elements of the main node only
+* ``--mcs_all <bool>`` (default ``false``\ ) -- Specify whether all the Minimal Cut Sets must be computed or just one
+* ``--mcs_max_cardinality <int>`` (default ``-1``\ ) -- Only search for MCS of cardinality lower or equal to this parameter. If ``-1``, all cardinalities will be considered
```

Comment:
Only search for MCSes of cardinality lower or equal to this parameter. If ``-1``, all MCSes are computed